examples/ticks_and_spines/tick_transform_formatter.py

```diff
@@ -0,0 +1,124 @@
+        """
+        Sets up the transformation such that `in_start` gets mapped to
+        `out_start` and `in_end` gets mapped to `out_end`. The following
+        shortcuts apply when only some of the inputs are specified:
+
+          - none: no-op
+          - in_start: translation to zero
+          - out_start: translation from zero
+          - in_end: scaling to one (divide input by in_end)
+          - out_end: scaling from one (multiply input by in_end)
+          - in_start, out_start: translation
+          - in_end, out_end: scaling (in_start and out_start zero)
+          - in_start, out_end: in_end=out_end, out_start=0
+          - in_end, out_start: in_start=0, out_end=in_end
+
+        Based on the following rules:
+
+          - start missing: set start to zero
+          - both ends are missing: set ranges to 1.0
+          - one end is missing: set it to the other end
+        """
+        if in_end is not None:
+            in_scale = in_end - in_start
+        elif out_end is not None:
+            in_scale = out_end - in_start
+        else:
+            in_scale = 1.0
+
+        if out_end is not None:
+            out_scale = out_end - out_start
+        elif in_end is not None:
+            out_scale = in_end - out_start
+        else:
+            out_scale = 1.0
+
+        self._scale = out_scale / in_scale
+        self._offset = out_start - self._scale * in_start
+
+    def __call__(self, x):
+        """
+        Transforms the input value `x` according to the rule set up in
+        `__init__`.
+        """
+        return x * self._scale + self._offset
+
+# X-data
+temp_C = np.arange(-5.0, 5.1, 0.25)
+# Y-data
+counts = 15.0 * np.exp(-temp_C**2 / 25)
+# Add some noise
+counts += np.random.normal(scale=4.0, size=counts.shape)
+if counts.min() < 0:
+    counts += counts.min()
+
+fig, ax1 = plt.subplots()
+ax2 = fig.add_subplot(111, sharex=ax1, sharey=ax1, frameon=False)
+
+ax1.plot(temp_C, counts, drawstyle='steps-mid')
+
+ax1.xaxis.set_major_formatter(StrMethodFormatter('{x:0.2f}'))
+
+# This step is necessary to allow the shared x-axes to have different
+# Formatter and Locator objects.
+ax2.xaxis.major = Ticker()
+# 0C -> 491.67R (definition), -273.15C (0K)->0R (-491.67F)(definition)
+ax2.xaxis.set_major_locator(ax1.xaxis.get_major_locator())
+ax2.xaxis.set_major_formatter(
+        TransformFormatter(LinearTransform(in_start=-273.15, in_end=0,
+                                           out_end=491.67),
+                           StrMethodFormatter('{x:0.2f}')))
+
+# The y-axes share their locators and formatters, so only one needs to
+# be set
+ax1.yaxis.set_major_locator(MaxNLocator(integer=True))
+# Setting the transfrom to `int` will only alter the type, not the
+# actual value of the ticks
+ax1.yaxis.set_major_formatter(
+        TransformFormatter(int, StrMethodFormatter('{x:02X}')))
+
+ax1.set_xlabel('Temperature (\u00B0C)')
```

Comment:
@anntzer changed these to all use named entities; please change to `\N{DEGREE SIGN}`.